pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -89,29 +94,37 @@ func initSchedulerExtender(ctx context.Context) {
 	RegisterMetrics(metricsList...)
 
 	azurediskInformerFactory := azurediskInformers.NewSharedInformerFactory(kubeExtensionClientset, time.Second*30)
-	pvcInformerFactory := informers.NewSharedInformerFactory(kubeClientset, time.Second*30)
+	coreInformerFactory := informers.NewSharedInformerFactory(kubeClientset, time.Second*30)
 
 	azVolumeAttachmentInformer = azurediskInformerFactory.Disk().V1alpha1().AzVolumeAttachments()
 	azDriverNodeInformer = azurediskInformerFactory.Disk().V1alpha1().AzDriverNodes()
-	pvcInformer = pvcInformerFactory.Core().V1().PersistentVolumeClaims()
+	pvcInformer = coreInformerFactory.Core().V1().PersistentVolumeClaims()
+	pvInformer = coreInformerFactory.Core().V1().PersistentVolumes()
 
 	pvcInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		UpdateFunc: onPvcUpdate, // Using Update instead of Add because pvc is not bound on Add
 		DeleteFunc: onPvcDelete,
 	})
 
+	pvInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: onPVUpdate, // Using Update instead of Add because pvc is not bound on Add
+		DeleteFunc: onPVDelete,
+	})
+
 	klog.V(2).Infof("Starting informers.")
 	go azurediskInformerFactory.Start(ctx.Done())
-	go pvcInformerFactory.Start(ctx.Done())
+	go coreInformerFactory.Start(ctx.Done())
 
 	if !cache.WaitForCacheSync(
 		ctx.Done(),
 		azDriverNodeInformer.Informer().HasSynced,
 		azVolumeAttachmentInformer.Informer().HasSynced,
-		pvcInformer.Informer().HasSynced) {
+		pvcInformer.Informer().HasSynced,
+		pvInformer.Informer().HasSynced) {
 		klog.Fatalf("Failed to sync and populate the cache for informers %s ...", err)
 		os.Exit(1)
 	}
+	recover()
```

Comment:
process exits if the cache sync failed. What do we need the recover for? It will reach this step only if the cache sync happened.

Reply:
The recover is for re-populating the cacheMap in case schedulerExtender restarts.

Reply:
Yes, I don't think you need to do this manually. Caches and maps are populated on forced resync.

Reply:
resync only populates the informer cache and doesn't trigger any of the event handlers from create, update, and delete. So without the `recover` function, our local map of `pv` to `pvc` and `pvc` to `diskName` would be lost.

Reply:
It has been a while since I've looked at this, but I remember reading the source code to verify this behavior as there was no documentation around this. Let me try to find today and link it here. To my knowledge, forced sync should be updating the local maps as well. You could also try deleting an instance of the extender and verify the behavior.

Reply:
I have tested the behavior after removing the line and was able to confirm that the map was not getting updated. So it seems like the line would be necessary to repopulate our local map.

Reply:
https://github.com/kubernetes/kubernetes/blob/0cdd940ff6fbd8a4b22eb6f2eb1db0d25ddea6b9/staging/src/k8s.io/client-go/tools/cache/controller.go#L167